### ISSUE TITLE
Calendar A11yPass : Screen reader does not read 'Role' as "button" on date controls present in calendar window.

### ISSUE THREAD

**Issue by 999-Dev** [OfficeDev/office-ui-fabric-react#10793]
[Accessibility][Screen Reader : Calendar Control] : Screen reader does not announce the Role on date controls present in calendar window.

**User Experience:**
Screen reader uses will be impacted if screen reader will not announce 'Role' the date controls.
 
**Test Environment:**
OS: Version: RS5_Prerelease(1903 OS Build 18999.1)
URL: https://developer.microsoft.com/en-us/fabric#/controls/web/calendar
Edge Browser Version:
Microsoft Edge: 44.18999.1.0
Microsoft EdgeHTML: 18.18999
Chrome Browser Version:
Chrome: 77.0.3865.90 

**Usage:**
All.

**Repro Steps:**
1.Open https://developer.microsoft.com/en-us/fabric#/controls/web/calendar in Edge and turn on Narrator.
2. Navigate to 'Galleries and Pickers' button and activate it.
3. Navigate to 'Calendar' link and activate it.
4. Calendar pane will be appeared.
5. Navigate date controls under the calendar window and verify the issue.

**Actual Result:**
Screen reader does not announce the 'Role' of the date controls. 

**Expected Result:**
Screen reader should announce the 'Role' as 'Button' of the date controls.

**MAS Reference:**
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={a12f3779-bdf1-40c2-bbcc-0e314ea94858}

**Attachment:**
[Screen reader does not announce the role of the date control.zip](https://github.com/OfficeDev/office-ui-fabric-react/files/3717334/Screen.reader.does.not.announce.the.role.of.the.date.control.zip)
![Role is not announced by the screen reader](https://user-images.githubusercontent.com/55537167/66649330-7fa91300-ec4b-11e9-82db-3e72aa480997.png)

**Comment by dzearing** [comment]
@lorejoh12 can you validate if this is a real issue/concern? 

If so please adjust the labels accordingly. Thank you!

**Comment by lorejoh12** [comment]
It's a minor one. Right now we have the buttons in the grid with role="gridcell". According to the w3c standards, https://www.w3.org/TR/wai-aria-practices/examples/grid/dataGrids.html, just by using role=grid on the table, we shouldn't need the gridcell role at all, but if we don't use it, Edge with Narrator will not read out which row/column the cell you focus is on.

With some limited testing, if we try setting role=gridcell on the <td> itself and then role="button" on the button inside the cell, that seems to announce what we want (both that we're on a "button" and which row/column the button is in). My concern is that we needed that extra role=gridcell for something else that I'm missing at the moment, and by changing it back to button (which we had in the past) we'll have a regression. That being said, I can't see this fix being prioritized soon, although we can add it to our backlog.

**Comment by lorejoh12** [comment]
I dug into this a little and figured out why we were using role="gridcell" on the buttons. If you use role="button", Narrator will no longer announce the selected status, it ignores aria-selected=true on the selected date. I think that's a little more important than the role being technically wrong, so I'd suggest we not make this change.

**Comment by lorejoh12** [comment]
Following up, the W3 example for a datepicker also doesn't announce selected state: https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html

given that, probably better to have correct button semantics than announcing the selected state on buttons, which isn't supported:  https://www.w3.org/WAI/PF/aria/roles#button
